test_data/config.toml

```diff
@@ -775,3 +775,21 @@ myhello.command = """
 myhello.runtime-packages = [ "hello", "goodbye" ]
 ' | flox lock-manifest - > "$RESPONSE_FILE"
 '''
+
+[build.hello]
+files = ["build/hello/hello.c", "build/hello/.flox"]
+pre_cmd = '''
+    rm .flox/env/manifest.lock
+'''
+cmd = '''
+    # re-resolve and build
+    flox lock-manifest .flox/env/manifest.toml > .flox/env/manifest.lock
+'''
+post_cmd = '''
+    build_output_dir="$(dirname $RESPONSE_FILE)"
+    build_dir="$build_output_dir/hello"
+    mkdir -p "$build_dir"
+    cp -R .flox "$build_dir/.flox"
+    cp hello.c "$build_dir/hello.c"
+    mv "$RESPONSE_FILE" "$build_dir/hello.yaml"
+'''
```

Comment:
As a pattern at least the post_cmd looks pretty ad-hoc, as deviating behaviour is implemented in-line (and would need to be done so for further tests).
I appreciate having a separate category for the sake of clarity what the data represents. On the other hand i suspect that providing different behaviour via this config mostly just shifts boilerplate from the test to the test_data gen, negates the consistency of mock data and its generation.
I agree that the `env.build_*` gens are already a hack (smuggling a lockfile into the output, rather than resolution data), but creating more such workarounds at this end of the generation seems counterproductive in the longer run.

Reply:
Agree, though I'm not sure it's worth optimising right now, until we know that this new pattern proves successful.